### PR TITLE
Feature/create classes

### LINKED ISSUE #15: Create Destinations Class
Create:
- Destination.js
- Destination-Repository.js
- Destination-test.js
- sampleData-destination.js

### PR DESCRIPTION
**Pull request by VictoriaFC**
### CHANGES

Created:
- Traveler.js
- Traveler-test.js
- TravelerRepository.js
- TravelerRepository-test.js
- Trip.js
- Trip-test.js
- TripRepository.js
- TripRepository-test.js
- Destination.js
- Destination-test.js
- DestinationRepository.js
- DestinationRepository-test.js

Wrote all classes/constructors and tests for all current code.

Closes #15 #14 #3

### CODE REVIEW

File: src/Destination.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Destination {
+    constructor(destinationData) {
+        this.id = destinationData.id;
+        this.destination = destinationData.destination;
+        this.estimatedLodgingCostPerDay = destinationData.estimatedLodgingCostPerDay;
+        this.estimatedFlightCostPerPerson = destinationData.estimatedFlightCostPerPerson;
+        this.image = {
```
Comment (BlueNealis) [inline src/Destination.js:7]: this is interesting I didn't think to do it this way!

File: src/DestinationRepository.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import Destination from './Destination';
+
+class DestinationRepository {
+    constructor(destinationData) {
+			this.destinations = destinationData.map((destinationObj) => { return new Destination(destinationObj) });
+    }
```
Comment (BlueNealis) [inline src/DestinationRepository.js:6]: Oh nice you create the destination objects in here.  I wonder if me not doing it in here will cause me trouble!

File: src/data/sampleData-trip.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    {"id":3,"userID":3,"destinationID":22,"travelers":4,"date":"2022/05/22","duration":17,"status":"approved","suggestedActivities":[]},
+    {"id":4,"userID":43,"destinationID":14,"travelers":2,"date":"2022/02/25","duration":10,"status":"approved","suggestedActivities":[]},
+    {"id":5,"userID":42,"destinationID":29,"travelers":3,"date":"2022/04/30","duration":18,"status":"approved","suggestedActivities":[]},
+    {"id":6,"userID":29,"destinationID":35,"travelers":3,"date":"2022/06/29","duration":9,"status":"approved","suggestedActivities":[]},
+    {"id":7,"userID":37,"destinationID":17,"travelers":5,"date":"2022/5/28","duration":20,"status":"approved","suggestedActivities":[]},
+    {"id":8,"userID":36,"destinationID":39,"travelers":6,"date":"2022/02/07","duration":4,"status":"approved","suggestedActivities":[]},
+    {"id":9,"userID":24,"destinationID":19,"travelers":5,"date":"2022/12/19","duration":19,"status":"approved","suggestedActivities":[]},
+    {"id":10,"userID":9,"destinationID":50,"travelers":6,"date":"2022/07/23","duration":17,"status":"approved","suggestedActivities":[]}
+]
```
Comment (BlueNealis) [inline src/data/sampleData-trip.js:12]: nice! Separate DATA WOOOOOO

File: test/TripRepository-test.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import TripRepository from '../src/TripRepository';
+import Trip from '../src/Trip'; 
+import  tripData  from '../src/data/sampleData-trip';
+
+describe('Trip Repository', () => {
+	let newTrip;
+  beforeEach(() => {
+    newTrip = new TripRepository(tripData);
+  });
+  it('should be a function', function () {
```
Comment (BlueNealis) [inline test/TripRepository-test.js:11]: Nice use of hooks

**Review by BlueNealis — APPROVED** [review]

File: src/DestinationRepository.js (hunk 2 above)
Comment (VictoriaFC) [reply src/DestinationRepository.js:6]: I don't think it'll cause you any problems with how you have it! Just a different way to do it :)

**Pull request merged**